### PR TITLE
Extr pref doom train

### PR DESCRIPTION
**Pull request by ricvolpe**
I started refactoring the extract preferences techniques notebooks into python scripts so that we can use them with the GCP hypertune framework (here is why so many changes).

I still have a couple of things to do (enable using both TfExtractor and TorchExtractor from launch script, set up GCP Hyperptune) but I wanted to show the changes and have them reviewed early on, as to avoid submitting a merge request that is too big and overwhelming.

Here is an example command to use the scripts (I will also add a README later)
```
python3 extract_preferences.py --data_path gs://pref_extract_train_output/ppo_search_log_fix_1455626/10/exp_data_500.pkl --agent_path gs://pref_extract_train_output/ppo_search_log_fix_1455626/10/saved_model_cp_8000 --gin_file extractors/configs/tf.gin
```

### CODE REVIEW

**Review by MislavJuric — APPROVED** [review]

**Review by MislavJuric — APPROVED** [review]

File: extractors/torch_extractor.py [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+        
+        self.num_train = num_train
+        self.num_val = num_val
+        self.num_repeat = num_repeat
+        self.epochs = epochs
+        self.batch_size = batch_size
+        self.learning_rate = learning_rate
+        self.weight_decay = weight_decay
+                
+        np.random.seed(0)
```
Comment (wichersn) [inline extractors/torch_extractor.py:155]: I think it's better to not set the seed so we can see how consistent the results are.

File: extractors/torch_extractor.py [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+            x = random_obs_torch
+            for lx, f in enumerate(torch_layers):
+                x = f(x)
+                if lx < len(self.model.module_list) - 2:
+                    x = nn.functional.relu(x)
+            torch_out = x.detach().cpu().numpy()
+            tf_out = tf_layer(random_obs).numpy()
+            if len(tf_out.shape) > 2:
+                tf_out = np.rollaxis(tf_out, 3, 1)            
+            np.testing.assert_allclose(torch_out, tf_out, rtol=.1, atol=5)
```
Comment (wichersn) [inline extractors/torch_extractor.py:220]: Yeah, could be a bug if the ratio has to be so high

File: extractors/torch_extractor.py [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+                'trainAUC': train_aucs, 'testAUC': test_aucs}
+
+    def train(self, xs, ys):
+        
+        averaged_results = {}
+        for run_ix in range(self.num_repeat):
+            print(f'Train pass no. {run_ix+1}')
+            results = self.single_train_pass(xs, ys)       
+        
+        for res in results:
```
Comment (wichersn) [inline extractors/torch_extractor.py:367]: For clarity, fix the indentation bug in this PR instead of the next one.

File: extractors/torch_extractor.py (hunk 2 above)
Comment (ricvolpe) [reply extractors/torch_extractor.py:220]: Note that from the training results there does not seems to be a bug.
